src/HttpClientRequestAdapter.cs

```diff
@@ -470,51 +469,85 @@ private async Task<HttpResponseMessage> GetHttpResponseMessage(RequestInformatio
             activityForAttributes?.SetTag("http.flavor", $"{response.Version.Major}.{response.Version.Minor}");
 
             return await RetryCAEResponseIfRequired(response, requestInfo, cancellationToken, claims, activityForAttributes).ConfigureAwait(false);
         }
+
         private static readonly Regex caeValueRegex = new("\"([^\"]*)\"", RegexOptions.Compiled, TimeSpan.FromMilliseconds(100));
+
         /// <summary>
         /// The key for the event raised by tracing when an authentication challenge is received
         /// </summary>
         public const string AuthenticateChallengedEventKey = "com.microsoft.kiota.authenticate_challenge_received";
+
         private async Task<HttpResponseMessage> RetryCAEResponseIfRequired(HttpResponseMessage response, RequestInformation requestInfo, CancellationToken cancellationToken, string? claims, Activity? activityForAttributes)
         {
             using var span = activitySource?.StartActivity(nameof(RetryCAEResponseIfRequired));
             if(response.StatusCode == HttpStatusCode.Unauthorized &&
                 string.IsNullOrEmpty(claims) && // avoid infinite loop, we only retry once
-                (requestInfo.Content?.CanSeek ?? true) &&
-                response.Headers.WwwAuthenticate?.FirstOrDefault(filterAuthHeader) is AuthenticationHeaderValue authHeader &&
-                authHeader.Parameter?.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries)
-                                                .Select(static x => x.Trim())
-                                                .FirstOrDefault(static x => x.StartsWith(ClaimsKey, StringComparison.OrdinalIgnoreCase)) is string rawResponseClaims &&
-                    caeValueRegex.Match(rawResponseClaims) is Match claimsMatch &&
-                    claimsMatch.Groups.Count > 1 &&
-                    claimsMatch.Groups[1].Value is string responseClaims)
+                (requestInfo.Content?.CanSeek ?? true))
             {
-                span?.AddEvent(new ActivityEvent(AuthenticateChallengedEventKey));
-                activityForAttributes?.SetTag("http.retry_count", 1);
-                requestInfo.Content?.Seek(0, SeekOrigin.Begin);
-                await DrainAsync(response, cancellationToken).ConfigureAwait(false);
-                return await GetHttpResponseMessage(requestInfo, cancellationToken, activityForAttributes, responseClaims).ConfigureAwait(false);
+                AuthenticationHeaderValue? authHeader = null;
+                foreach(var header in response.Headers.WwwAuthenticate)
+                {
+                    if(filterAuthHeader(header))
+                    {
+                        authHeader = header;
+                        break;
+                    }
+                }
+
+                if(authHeader is not null)
+                {
+                    var authHeaderParameters = authHeader.Parameter?.Split([','], StringSplitOptions.RemoveEmptyEntries);

```

Comment:
net standard compatibility
```suggestion
                    var authHeaderParameters = authHeader.Parameter?.Split(new[]{','}, StringSplitOptions.RemoveEmptyEntries);

```